File: JINSEOP/프로그래머스_WEEK12/멀리뛰기.cpp
#include <string>
#include <vector>
using namespace std;

long long DP[2001];

void init_DP() {
    DP[0] = 0;
    DP[1] = 1;
    DP[2] = 2;
}

void do_DP(int n) {
    init_DP();
    for (int i = 3; i <= n; i++)
        DP[i] = (DP[i - 1] + DP[i - 2]) % 1234567;
}

long long solution(int n) {
    long long answer = 0;

    do_DP(n);
    answer = DP[n];
    return answer;
}

File: DONGUK/프로그래머스_WEEK_12/멀리뛰기.py
def solution(n):
    answer = 0
    
    dp = [0] * (1+n)
    if n == 1:
        return 1
    
    dp[1] = 1
    dp[2] = 2
    
    if n >= 3:
        for i in range(3, n+1):
            dp[i] = (dp[i-1] + dp[i-2]) % 1234567
    
    answer = dp[n] % 1234567
    return answer